src/Bicep.Core/TypeSystem/DeployTimeConstantVisitor.cs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.TypeSystem
+{
+    /// <summary>
+    /// Visitor used to collect errors caused by property assignments to run-time values when that is not allowed
+    /// </summary>
+    public sealed class DeployTimeConstantVisitor : SyntaxVisitor
+    {
+
+        // These constants currently resolve to the same name.
+        private readonly HashSet<string> DeployTimeConstantKeys = new HashSet<string>()
+        { 
+            LanguageConstants.ResourceNamePropertyName,
+            LanguageConstants.ModuleNamePropertyName
+        }; 
+
```

Comment:
I think you should be able to check for `TypePropertyFlags.SkipInlining` on the type rather than specifying the names of properties to check.

Reply:
Yet we check for `! TypePropertyFlags.SkipInlining` (not). I've renamed it DeployTimeConstant as you've suggested but we may be checking for the opposite...

Reply:
Verified that what I said is **not** true... This is done